Offline/OfflinePlayer/objc/OfflinePlayer/VideosViewController.m

```diff
@@ -0,0 +1,1370 @@
+        self.playerView.playbackController = playbackController;
+    }
+}
+
+- (void)showWarnings
+{
+    // Simulator warning
+#if (TARGET_OS_SIMULATOR)
+    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Reminder..."
+                                                                   message:@"FairPlay videos won't display in a simulator."
+                                                            preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
+                                                          handler:^(UIAlertAction * action) {}];
+    
+    [alert addAction:defaultAction];
+    [self presentViewController:alert animated:YES completion:nil];
+#endif
+    
+    // Account credentials warning
+    if (kDynamicDeliveryAccountID.length == 0)
+    {
+        UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Invalid account information"
+                                                                       message:@"Don't forget to enter your account information at the top of VideosViewController.m."
+                                                                preferredStyle:UIAlertControllerStyleAlert];
+        UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
+                                                              handler:^(UIAlertAction * action) {}];
+        
+        [alert addAction:defaultAction];
+        [self presentViewController:alert animated:YES completion:nil];
+        
+    }
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.downloadSizeDictionary = [NSMutableDictionary dictionary];
+
+    NSDictionary *optionsDictionary =
+    @{
+      kBCOVOfflineVideoManagerAllowsCellularDownloadKey: @(NO),
+      kBCOVOfflineVideoManagerAllowsCellularPlaybackKey: @(NO),
+      kBCOVOfflineVideoManagerAllowsCellularAnalyticsKey: @(NO)
+      };
+    [BCOVOfflineVideoManager initializeOfflineVideoManagerWithDelegate:self
+                                                               options:optionsDictionary];
+    self.offlineVideoManager = BCOVOfflineVideoManager.sharedManager;
+    gVideosViewController = self;
+
+    NSLog(@"Using Brightcove Native Player SDK version %@", BCOVPlayerSDKManager.version);
+    
+    self.tabBarController = (UITabBarController*)self.parentViewController;
+    
+    self.videosTableView.dataSource = self;
+    self.videosTableView.delegate = self;
+    [self.videosTableView setContentInset:UIEdgeInsetsMake(0, 0, 8, 0)];
+ 
+    // Add a refresh control to the table view
+    {
+        self.refreshControl = [[UIRefreshControl alloc] init];
+        [self.refreshControl addTarget:self action:@selector(handleTableRefresh:) forControlEvents:UIControlEventValueChanged];
+        
+        if (floor(NSFoundationVersionNumber) <= NSFoundationVersionNumber_iOS_9_x_Max)
```

Comment:
Should probably switch this to use @available - looks out of place.

Reply:
```
    if (@available (iOS 10.0, *))
    {
        // iOS 10 and later: proper refresh control support
        self.videosTableView.refreshControl = self.refreshControl;
    }
    else
    {
        // iOS 9 and earlier: background view is the refresh control
        self.videosTableView.backgroundView = self.refreshControl;
    }
```